fieldservice_maintenance/models/fsm_order.py

```diff
@@ -11,30 +11,30 @@ class FSMOrder(models.Model):
     request_id = fields.Many2one('maintenance.request',
                                  string='Maintenance Request')
 
-    # @api.model
-    # def create(self, vals):
-    #     # if FSM order with type maintenance is create then
-    #     # create maintenance request
-    #     order = super(FSMOrder, self).create(vals)
-    #     if order.type == 'maintenance':
-    #         employee_rec = self.env['hr.employee'].search(
-    #             [('user_id', '=', self.env.uid)], limit=1)
-    #         if order.equipment_id and not order.request_id:
-    #             equipment = order.equipment_id
-    #             team_id = equipment.maintenance_equipment_id and\
-    #                 equipment.maintenance_equipment_id.maintenance_team_id.id
-    #             request_id = self.env['maintenance.request'].with_context(
-    #                 fsm_order=True).create({
-    #                     'name': order.name or '',
-    #                     'employee_id': employee_rec.id,
-    #                     'equipment_id':
-    #                         equipment.maintenance_equipment_id.id,
-    #                     'category_id': equipment.category_id.id,
-    #                     'request_date': fields.Date.context_today(self),
-    #                     'maintenance_type': 'corrective',
-    #                     'maintenance_team_id': team_id,
-    #                     'schedule_date': order.request_early,
-    #                     'description': order.description
-    #                 })
-    #             order.request_id = request_id
-    #     return order
+    @api.model
+    def create(self, vals):
+        # if FSM order with type maintenance is create then
+        # create maintenance request
+        order = super(FSMOrder, self).create(vals)
+        if order.type == 'maintenance':
```

Comment:
@bodedra The type is now a many2one field. We need to use the external id.

Reply:
@max3903 Yes. I first uncomment lines and correct the logic.